### PR TITLE
Fix for undefined index REQUEST_METHOD for some background jobs

### LINKED ISSUE #4286: Publicize: Undefined index: REQUEST_METHOD
#### Steps to reproduce the issue

* Configure Publicize to automatically publicize when a post is created
* Create a cron job to publish a post
* Run the cron job using [Cavalcade](https://github.com/humanmade/Cavalcade) (or any system that doesn't set `REQUEST_METHOD`)

#### What happened

The following notice was generated:

```
E_NOTICE: /home/wporg/public_html/wp-content/plugins/jetpack/modules/publicize/publicize.php:329 - Undefined index: REQUEST_METHOD*
```

#### Example Cron Job

```php
class WP_Make_Community_Daily_Prompt {
    public static function load() {
        if ( ! wp_next_scheduled( 'wp_make_community_daily_prompt' ) )
            wp_schedule_single_event( strtotime( 'tomorrow 4 am utc' ), 'wp_make_community_daily_prompt' );

        add_action( 'wp_make_community_daily_prompt', array( __CLASS__, 'prompt' ) );
    }

    public static function prompt() {
        $post_id = wp_insert_post( array(
            'post_content' => sprintf( '%s #daily-updates', gmdate( 'l', time() ) ),
            'post_status' => 'publish',
            'post_author' => 7694169, // wordcamp
        ) );

        set_post_format( $post_id, 'status' );
    }
}

add_action( 'muplugins_loaded', array( 'WP_Make_Community_Daily_Prompt', 'load' ) );
```

### PR DESCRIPTION
**Pull request by gravityrail**
Fixes #4286.

Just checks that REQUEST_METHOD isset on $_SERVER to avoid a PHP notice.

### CODE REVIEW

File: modules/publicize/publicize.php [hunk 1]
```diff
@@ -326,7 +326,11 @@ function save_meta( $post_id, $post ) {
 		}
 
 		// Did this request happen via wp-admin?
-		$from_web = 'post' == strtolower( $_SERVER['REQUEST_METHOD'] ) && isset( $_POST[$this->ADMIN_PAGE] );
+		$from_web = isset( $_SERVER['REQUEST_METHOD'] )
+			&&
+			'post' == strtolower( $_SERVER['REQUEST_METHOD'] )
```
Comment (lezama) [inline modules/publicize/publicize.php]: let's add a strict comparison since we are here

File: modules/publicize/publicize.php [hunk 2]
```diff
@@ -326,7 +326,11 @@ function save_meta( $post_id, $post ) {
 		}
 
 		// Did this request happen via wp-admin?
-		$from_web = 'post' == strtolower( $_SERVER['REQUEST_METHOD'] ) && isset( $_POST[$this->ADMIN_PAGE] );
+		$from_web = isset( $_SERVER['REQUEST_METHOD'] )
+			&&
+			'post' == strtolower( $_SERVER['REQUEST_METHOD'] ) 
+			&& 
+			isset( $_POST[$this->ADMIN_PAGE] );
```
Comment (lezama) [inline modules/publicize/publicize.php]: let's add spaces around `$this->ADMIN_PAGE`

**Comment by lezama** [comment]
Looks good to me 👍 🚢

**Comment by gravityrail** [comment]
Merged in 4.2

**Pull request merged**